### PR TITLE
replace json marshalling with protojson

### PR DESCRIPTION
**Pull request by cendhu**
Signed-off-by: senthil <cendhu@gmail.com>

### CODE REVIEW

File: pkg/bcdb/mocks/rest_client.go [hunk 1]
```diff
@@ -42,11 +42,11 @@ func (_m *RestClient) Query(ctx context.Context, endpoint string, httpMethod str
 }
 
 // Submit provides a mock function with given fields: ctx, endpoint, msg, serverTimeout
-func (_m *RestClient) Submit(ctx context.Context, endpoint string, msg protoiface.MessageV1, serverTimeout time.Duration) (*http.Response, error) {
+func (_m *RestClient) Submit(ctx context.Context, endpoint string, msg protoreflect.ProtoMessage, serverTimeout time.Duration) (*http.Response, error) {
```
Comment (tock-ibm) [inline pkg/bcdb/mocks/rest_client.go:45]: use the proto.Message type instead, which aliases this type.

File: pkg/bcdb/mocks/rest_client.go [hunk 2]
```diff
@@ -42,11 +42,11 @@ func (_m *RestClient) Query(ctx context.Context, endpoint string, httpMethod str
...
 	ret := _m.Called(ctx, endpoint, msg, serverTimeout)
 
 	var r0 *http.Response
-	if rf, ok := ret.Get(0).(func(context.Context, string, protoiface.MessageV1, time.Duration) *http.Response); ok {
+	if rf, ok := ret.Get(0).(func(context.Context, string, protoreflect.ProtoMessage, time.Duration) *http.Response); ok {
```
Comment (tock-ibm) [inline pkg/bcdb/mocks/rest_client.go:49]: use the proto.Message type instead, which aliases this type.

File: pkg/bcdb/mocks/rest_client.go [hunk 3]
```diff
@@ -55,7 +55,7 @@ func (_m *RestClient) Submit(ctx context.Context, endpoint string, msg protoifac
 	}
 
 	var r1 error
-	if rf, ok := ret.Get(1).(func(context.Context, string, protoiface.MessageV1, time.Duration) error); ok {
+	if rf, ok := ret.Get(1).(func(context.Context, string, protoreflect.ProtoMessage, time.Duration) error); ok {
```
Comment (tock-ibm) [inline pkg/bcdb/mocks/rest_client.go:58]: use the proto.Message type instead, which aliases this type.

Comment (cendhu) [reply pkg/bcdb/mocks/rest_client.go:58]: auto generated code by mockery.

File: pkg/bcdb/mocks/rest_client.go (hunk 2 above)
Comment (cendhu) [reply pkg/bcdb/mocks/rest_client.go:49]: auto generated code by mockery.

File: pkg/bcdb/mocks/rest_client.go (hunk 1 above)
Comment (cendhu) [reply pkg/bcdb/mocks/rest_client.go:45]: auto generated code by mockery.

**Comment by cendhu** [comment]
@tock-ibm As rest_client.go is auto generated code, I have not updated it and left it as is.

**Comment by tock-ibm** [comment]
> @tock-ibm As rest_client.go is auto generated code, I have not updated it and left it as is.

@cendhu True! I missed that.

**Review by tock-ibm — APPROVED** [review]

**Pull request merged**